llvm-generator/lib/Minicute/Transpilers/Generator.hs

```diff
@@ -46,60 +99,33 @@ generateMachineCodeE = go []
       _ <- call (AST.ConstantOperand constantUpdateNodeNInteger) [(vName, []), (nName, [])]
 
       go vStack insts
-    go _ [IReturn] = do
-      bName <- load (AST.ConstantOperand constantAddrBasePointer) 0
-      bName' <- gep bName [operandInt 32 0]
-      store bName' 0 (AST.ConstantOperand constantAddrStackPointer)
-      retVoid
-    go _ _ = error "generateMachineCodeE: Not yet implemented"
-
--- *
--- We should extract or export the following definitions
--- since tests use them too.
-operandInt :: Word32 -> Integer -> AST.Operand
-operandInt w n = AST.ConstantOperand (ASTC.Int w n)
-
-constantUpdateNodeNInteger :: ASTC.Constant
-constantUpdateNodeNInteger = ASTC.GlobalReference typeUpdateNodeNInteger "minicute_update_node_NInteger"
-
-constantAddrStackPointer :: ASTC.Constant
-constantAddrStackPointer = ASTC.GlobalReference typeInt8PtrPtrPtr "asp"
-
-constantAddrBasePointer :: ASTC.Constant
-constantAddrBasePointer = ASTC.GlobalReference typeInt8PtrPtrPtr "abp"
-
-constantNodeHeapPointer :: ASTC.Constant
-constantNodeHeapPointer = ASTC.GlobalReference typeInt8PtrPtr "nhp"
-
-typeUpdateNodeNInteger :: ASTT.Type
-typeUpdateNodeNInteger = ASTT.FunctionType ASTT.void [typeInt32, typeInt8Ptr] False
-
-typeNodeNIntegerPtr :: ASTT.Type
-typeNodeNIntegerPtr = ASTT.ptr typeNodeNInteger
 
-typeNodeNInteger :: ASTT.Type
-typeNodeNInteger = ASTT.NamedTypeReference "node.NInteger"
+    go _ [IUnwind] = do
+      _ <- call (AST.ConstantOperand constantUtilUnwind) []
 
-typeInt8PtrPtrPtr :: ASTT.Type
-typeInt8PtrPtrPtr = ASTT.ptr typeInt8PtrPtr
-
-typeInt8PtrPtr :: ASTT.Type
-typeInt8PtrPtr = ASTT.ptr typeInt8Ptr
+      retVoid
 
-typeInt8Ptr :: ASTT.Type
-typeInt8Ptr = ASTT.ptr typeInt8
+    go vStack (IEval : insts@(_ : _)) = do
+      evalBody
 
-typeInt8 :: ASTT.Type
-typeInt8 = ASTT.i8
+      go vStack insts
```

Comment:
Check `vStack`. It might be reset after `IEval`.

Reply:
It should preserve the `vStack` since it will be reset in `minicute__util__unwind`.